src/naive_bayes/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::error::Failed;
+use crate::linalg::Matrix;
+use crate::math::num::RealNumber;
+pub trait NBDistribution {
+    fn fit<T: RealNumber, M: Matrix<T>>(&self, x: &M, y: M::RowVector) -> Self;
```

Comment:
Let's make it static
```suggestion
    fn fit<T: RealNumber, M: Matrix<T>>(x: &M, y: M::RowVector) -> Self;
```